src/LightSpeed/userFunction.js

```diff
@@ -53,33 +53,100 @@ const COMMON_SCOPE = {
     },
 };
 
-const COMMON_SPAWN_SCOPE = {
-    // Keep this in sync with SpawnScope class in
-    // hack-toy-apps/com.endlessm.LightSpeed/userScope.js
-    ticksSinceSpawn: 0,
+// Set method that checks type for number properties of objects
+function propNumFactory(obj, propName, valName) {
+    Object.defineProperty(obj, propName, {
+        set(newVal) {
+            if (Number.isNaN(Number(newVal)))
+                throw new TypeError(`${propName} must be a number.`);
+            else if (Number(newVal) < 0)
+                throw new RangeError(`${propName} must be a positive number.`);
+            valName = newVal; // eslint-disable-line no-param-reassign
+        },
+        get() {
+            return valName;
+        },
+    });
+}
+
+const NUMTYPE_HANDLER = {
+    set(obj, prop, value) {
+        if (prop in obj) {
+            if (Number.isNaN(Number(value)))
+                throw new TypeError(`Value for ${prop} must be a number.`);
+            else
+                return Reflect.set(obj, prop, value);
+        } else {
+            throw new TypeError(`Unknown property ${prop}.`);
+        }
+    },
+
+    get(target, name, receiver) {
+        if (name in target)
+            return Reflect.get(target, name, receiver);
+
+        throw new TypeError(`Unknown property ${name}`);
+    },
 };
 
+// Proxy for position values
+const POSITION = {x: 0, y: 0};
+const POS_PROXY = new Proxy(POSITION, NUMTYPE_HANDLER);
```

Comment:
:+1: your idea of using proxies and/or getters/setters for these where possible, instead of checking the values at the end of the script. It makes it possible to mark the errors as coming from the correct line. I didn't think of that.